xdem/spatialstats.py

```diff
@@ -280,10 +282,233 @@ def interp_nd_binning(df: pd.DataFrame, list_var_names: Union[str,list[str]], st
     return interp_fun
 
 
+def two_step_standardization(dvalues: np.ndarray, list_var: Iterable[np.ndarray],
+                unscaled_error_fun: Callable[[tuple[np.ndarray, ...]], np.ndarray],
+                spread_statistic: Callable = nmad,
+                fac_spread_outliers: float | None = 7
+                ) -> tuple[np.ndarray, Callable[[tuple[np.ndarray, ...]], np.ndarray]]:
+    """
+    Standardize the proxy differenced values using the modelled heteroscedasticity, re-scaled to the spread statistic,
```

Comment:
I'm wondering. Should it not just be an option of `interp_nd_binning` to return a function for which the spread is 1 when applied to a given sample? That would be easier than having this additional function.
One could possibly have a single function that does the binning, interpolation and second standardization, no?

Reply:
Just saw the function `estimate_model_heteroscedasticity` below ! 👏

Reply:
Maybe you missed my very first comment, regarding the two step standardization?